httputil/httputil.go

```diff
@@ -16,12 +17,24 @@ type Consumer interface {
 	String() string
 }
 
+type stateMachine interface {
+	State() (*statemachine.State, time.Time)
+}
+
+type Task struct {
+	ID       string     `json:"id"`
+	Started  time.Time  `json:"started"`
+	Stopped  *time.Time `json:"stopped,omitempty"`
+	State    string     `json:"state,omitempty"`
+	Modified *time.Time `json:"modified,omitempty"`
```

Comment:
Yeah... it's a bit weirdly named since really it's the last time the statemachine looped. One of those times where I spent 30s trying to think of a more accurate name that wasn't 50 characters long and gave up. :)